src/DocBlock/Annotation.php

```diff
@@ -147,7 +160,25 @@ public function getTag()
      */
     public function getTypes()
     {
-        return explode('|', $this->getTypesContent());
+        if (null === $this->types) {
+            $this->types = explode('|', $this->getTypesContent());
```

Comment:
nice catch, it's from old implementation when i was compering old behaviour vs new one